### DISCUSSION TITLE
Graph issue after upgrading to 1.3 (from 1.2b6)

### DISCUSSION THREAD

**Discussion by Brezentrager** [poblabs/weewx-belchertown#868] [Q&A]
In my graphs.conf, there are definitions of two quite similar charts. One displaying daily values, one showing monthly sums.

    [[Sonnenstunden]]
        title = Sonnenstunden
        aggregate_type = sum
        aggregate_interval = 86400  # 1 Tag
        type = column
        [[[sunshineTime]]]
            observation_type = sunshineTime / 60
            name = Sonnenstunden pro Tag
            yAxis_min = 0
            yAxis_label = "Stunden Sonnenschein pro Tag"
            color = "#f7f2b4"

    [[SonnenstundenMonatlich]]
        title = Sonnenstunden
        aggregate_type = sum
        xAxis_groupby = month
        xAxis_categories = 'Jan', 'Feb', 'Mär', 'Apr', 'Mai', 'Jun', 'Jul', 'Aug', 'Sep', 'Okt', 'Nov', 'Dez'
        type = column
        [[[sunshineTime]]]
            observation_type = sunshineTime / 60
            name = Sonnenstunden pro Monat
            yAxis_min = 0
            yAxis_label = "Stunden Sonnenschein pro Monat"
            color = "#f7f2b4"

Before upgrading to 1.3, both charts had worked. But now the second chart fails.
The report containing this chart is not produced - nor are all the following reports in graphs.conf

The issue obviously has to do with the xAxis grouping. When commenting out those two lines, the chart works. And when deleting the calculation for the observation_type, the charts is working, too.

So, with the latest version, the combination of xAxis grouping and calculations for the observation_type seems to have an issue.

**Comment by Brezentrager** [comment]
This is the error log:


Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine: Caught unrecoverable exception in generator 'user.belchertown.HighchartsJsonGenerator'
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  near "/": syntax error
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  Traceback (most recent call last):
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weedb/sqlite.py", line 39, in guarded_fn
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      return fn(*args, **kwargs)
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weedb/sqlite.py", line 233, in execute
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      return sqlite3.Cursor.execute(self, *args, **kwargs)
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  sqlite3.OperationalError: near "/": syntax error
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  During handling of the above exception, another exception occurred:
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  Traceback (most recent call last):
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 197, in run
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      obj.start()
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 385, in start
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      self.run()
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/belchertown.py", line 2771, in run
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      special_target_unit
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/belchertown.py", line 3694, in get_observation_data
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      for row in query:
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/manager.py", line 598, in genSql
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      for _row in _cursor.execute(sql, sqlargs):
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weedb/sqlite.py", line 53, in guarded_fn
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****      raise weedb.OperationalError(e)
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  weedb.OperationalError: near "/": syntax error
Mar 10 23:20:31 raspberrypi weewx[15904] ERROR weewx.reportengine:         ****  Generator terminated

**Comment by roe-dl** [comment]
I would like to suggest that it's time to replace the SQL statements in `belchertown.py` by using the XTYPE functions, especially `get_series` and `get_aggregation`.
